scipp/whats-new-0.5.0.ipynb

```diff
@@ -0,0 +1,342 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.scalar(17)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All of these also take keyword arguments.\n",
+    "Note that we can still support creating scalars by multiplying with a unit:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "1.2 * sc.units.m"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Label-based indexing\n",
+    "\n",
+    "[Label-based indexing](https://scipp.github.io/user-guide/slicing.html#Label-based-indexing)\n",
+    "\n",
+    "- Works just like position indexing (with integers).\n",
+    "- Use a scalar variable as index (instead of integer) to use label-based indexing\n",
+    "- Works with single values as well as slices (`:` notation)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.to_html(hist['spectrum', sc.scalar(1000)])\n",
+    "sc.to_html(hist['tof', 1000*sc.units.us]) # selects bin containing this value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset items and masks\n",
+    "\n",
+    "- Previously masks for global for a dataset, masking items individually was not supported.\n",
+    "- Applications require item-specific masks.\n",
+    "- Removed support for dataset masks, only item masks are supported now, since having both causes conceptual problems.\n",
+    "- See [ADR0001](https://scipp.github.io/developer/adr/0001-remove-dataset-masking-support.html) for details.\n",
+    "- Common masks must be set individually."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hist = sc.histogram(data, tof_edges)\n",
+    "ds = sc.Dataset({'sample1':hist, 'sample2':hist+hist})\n",
+    "tof_mask = ds.coords['tof']['tof',:-1] < 3333*sc.units.us\n",
+    "ds['sample1'].masks['tof'] = tof_mask\n",
+    "ds['sample2'].masks['tof'] = tof_mask\n",
+    "ds['sample2'].masks['tof2'] = ~tof_mask\n",
+    "ds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## HDF5 I/O\n",
+    "\n",
+    "Scipp can now [write and read its data structures](https://scipp.github.io/user-guide/reading-and-writing-files.html) using `h5py`.\n",
+    "- At this point we do not make any promises about compatibilty of the file layout with future version, so we do not encourage usage for storing data long-term.\n",
+    "- Loading event data via Mantid converter is quite slow, so loading raw data once and storing natively can provide a UX boost when working on debugging a reduction workflow in a notebook.\n",
+    "- Note the warning message about `PyObject`, since we cannot write Mantid's `Sample` object."
```

Comment:
Aside from this caveat, the serialization/deserialization abilities are complete to my knowledge. Should we emphasise this. In other words, this isn't that scipp can save some of it's data under special circumstances, it can save/load everything excepting `PyObject`.